test/test_client.py

```diff
@@ -1601,6 +1601,19 @@ def test_network_error_message(self):
             with self.assertRaisesRegex(AutoReconnect, expected):
                 client.pymongo_test.test.find_one({})
 
+    def test_service_name_from_kwargs(self):
+        client = MongoClient(
+            'mongodb+srv://user:password@test22.test.build.10gen.cc',
+            srvServiceName='customname', connect=False)
+        self.assertEqual(client._topology_settings._srv_service_name,
+                         'customname')
+        client = MongoClient(
+             'mongodb+srv://user:password@test22.test.build.10gen.cc'
+             '/?srvServiceName=customname',
+             connect=False)
+        self.assertEqual(client._topology_settings._srv_service_name,
+                         'customname')
```

Comment:
We need a test which actually checks that the kwarg overrides the URI. Like this:
```python
        client = MongoClient(
             'mongodb+srv://user:password@test22.test.build.10gen.cc'
             '/?srvServiceName=shouldBeOverridden',
             srvServiceName='customname',
             connect=False)
        self.assertEqual(client._topology_settings._srv_service_name,
                         'customname')
```